app/schemas/user.js

```diff
@@ -0,0 +1,42 @@
+const { User } = require('../models');
+
+exports.user_schema = {
+  email: {
+    in: ['body'],
+    matches: {
+      options: /^[a-zA-Z0-9]+@wolox.([A-Za-z])+$/,
+      errorMessage: 'The email must have the wolox domain'
+    },
+    custom: {
+      options: value =>
+        User.findAll({ where: { email: value } }).then(users =>
+          users.length > 0
+            ? Promise.reject(Error('Email must be unique'))
```

Comment:
return the custom error we use

Reply:
why do you do a promise reject here while in the other validations you return an errorMessage?

Reply:
The response is using the Wolox error object, the custom validator schema expects a promise rejection for handling errors if the operation is async https://express-validator.github.io/docs/custom-validators-sanitizers.html. If i use the wolox error message in the promise rejection, the response message would be a nested object with two wolox error object.

Reply:
Mmm I dont think so. That error seems to be javascript Error object

Reply:
True, but the middleware is parsing the message and sending it as a wolox error message